src/component/todo-list/todo-list.ts

```diff
@@ -0,0 +1,109 @@
+import '../todo-item/todo-item';
+import { todoStore } from '../../utils/todo-store';
+import { filterStore } from '../../utils/filter-store';
+import { Todo } from '../../models/todo';
+import { FilterType } from '../../models/filter';
+
+export class TodoList extends HTMLElement {
+  private todoStore = todoStore;
+  private filterStore = filterStore;
+  private filter: FilterType = 'all';
+  private todoListData: Todo[] = [];
+  private sortFlag: boolean = true;
+
+  private updateTodoList = () => {
+    const todos = this.todoStore.getState();
+    this.filter = this.filterStore.getState();
+    this.todoListData = todos.filter(todo => {
+      if (this.filter === 'all') return true;
+      if (this.filter === 'active') return !todo.done;
+      if (this.filter === 'completed') return todo.done;
+      return true;
+    });
+
+    this.render();
+
+    this.dispatchEvent(
+      new CustomEvent('list-render', {
+        bubbles: true
+      })
+    );
+  };
+
+  constructor(sort?: boolean) {
+    super();
+    this.todoStore.subscribe(this.updateTodoList);
+    this.filterStore.subscribe(this.updateTodoList);
+    const sortFlag = sort || this.getAttribute('sort');
+    this.sortFlag = sortFlag === 'true' || sortFlag === true ? true : false;
+  }
+
+  connectedCallback(): void {
+    this.updateTodoList();
+    this.addEventListener('dnd-mouseup', this.onMouseUp);
+  }
+
+  private render(): void {
+    this.innerHTML = `
+      <style>
+        :host {
+          display: block;
+          width: 100%;
+        }
+        ul {
+          list-style: none;
+          padding: 0;
+        }
+      </style>
+      <ul class="todo-list">
+        ${this.viewItemList()}
+      </ul>
+    `;
+  }
+
+  disconnectedCallback(): void {
+    this.todoStore.unsubscribe(this.updateTodoList);
+    this.filterStore.unsubscribe(this.updateTodoList);
+  }
+
+  private viewItemList(): string {
+    if (this.todoListData.length < 1) return 'none~';
+
+    // dnd 아니면 정렬로직 적용
+    if (this.sortFlag) {
+      this.todoListData = this.todoListData.sort((a, b) => {
+        if (a.done === b.done) return b.id - a.id;
+        return a.done ? 1 : -1;
+      });
+    }
+
+    return this.todoListData
+      .map(
+        todo => `
+      <todo-item text="${todo.text}" done="${todo.done}" id="${todo.id}"></todo-item>
+    `
+      )
+      .join('');
+  }
+
+  private onMouseUp = (e: Event) => {
+    const ulTag = this.querySelector('ul.todo-list');
+    const { onDropzone, el } = (e as CustomEvent).detail;
+    if (!ulTag || !onDropzone || el !== ulTag) return;
+
+    this.changeSortToStore(ulTag as HTMLElement);
+  };
+  private changeSortToStore = (todoList: HTMLElement) => {
+    //현재 순서 가져온뒤
+    const order = Array.from(todoList.children).map(todo => {
+      return (todo as HTMLElement).id;
+    });
```

Comment:
```suggestion
    const order = Array.from(todoList.children).map(todo => todo as HTMLElement).id);
```
그리고 order 변수명에 해당값이 알맞지 않은거같네요.
가독성이 떨어지는데 확인부탁드립니다.